### PR TITLE
5485: Change expected 200 code to expected SUCCESSFUL family

### PR DESCRIPTION
**Pull request by kirmanak**

### CODE REVIEW

File: src/test/java/com/axibase/tsd/api/util/TestUtil.java [hunk 1]
```diff
@@ -46,6 +41,10 @@ public static String formatDate(Date date, String pattern) {
         return formatDate(date, pattern, getServerTimeZone());
     }
 
+    public static boolean responseFamilyEquals(final Response response, final Response.Status.Family family) {
```
Comment (raipc) [inline src/test/java/com/axibase/tsd/api/util/TestUtil.java]: I would better introduce the `responseFamily(Response response)` method and use it in tests. A separate method to test enums for equality is not needed

File: src/test/java/com/axibase/tsd/api/method/series/SeriesInsertTest.java [hunk 2]
```diff
@@ -408,66 +420,66 @@ public void testUrlNotFoundGetRequest0() throws Exception {
...
     }
 
     @Issue("2927")
     @Test
-    public void testUrlNotFoundOptionsRequest0() throws Exception {
+    public void testUrlNotFoundOptionsRequest0() {
         Response response = executeApiRequest(webTarget -> webTarget.path("*").request().options());
         response.bufferEntity();
-        assertEquals("Nonexistent url with /api/v1 options doesn't return 200", OK.getStatusCode(), response.getStatus());
+        assertSame("Nonexistent url without /api/v1 options doesn't return 200", Response.Status.Family.SUCCESSFUL, response.getStatusInfo().getFamily());
```
Comment (raipc) [inline src/test/java/com/axibase/tsd/api/method/series/SeriesInsertTest.java]: Please, parameterize these test cases.
Assertion message: "OPTIONS request to nonexistent URL doesn't return a successful status code"

File: src/test/java/com/axibase/tsd/api/util/TestUtil.java (hunk 1 above)
Comment (kirmanak) [reply src/test/java/com/axibase/tsd/api/util/TestUtil.java]: What does return the `responseFamily(Response response)` method?

Comment (kirmanak) [reply src/test/java/com/axibase/tsd/api/util/TestUtil.java]: @raipc What does return the `responseFamily(Response response)` method?

Comment (raipc) [reply src/test/java/com/axibase/tsd/api/util/TestUtil.java]: `Response.Status.Family`

Comment (kirmanak) [reply src/test/java/com/axibase/tsd/api/util/TestUtil.java]: @raipc 
So, you want to replace

```java
Util.responseFamilyEquals(response, Response.Status.Family.SUCCESSFUL)
```

with

```java
Util.responseFamily(response).equals(Response.Status.Family.SUCCESSFUL)
```

Did I understand it properly?

File: src/main/java/com/axibase/tsd/api/method/entitygroup/EntityGroupMethod.java [hunk 3]
```diff
@@ -53,10 +51,10 @@ public static void createOrReplaceEntityGroupCheck(EntityGroup entityGroup) thro
 
     public static boolean entityGroupExist(EntityGroup entityGroup) throws Exception {
         Response response = getEntityGroup(entityGroup.getName());
-        if (response.getStatus() == NOT_FOUND.getStatusCode()) {
+        if (Util.responseFamily(response).equals(Response.Status.Family.CLIENT_ERROR)) {
```
Comment (raipc) [inline src/main/java/com/axibase/tsd/api/method/entitygroup/EntityGroupMethod.java]: Please, replace all `Family` enum equality checks to `==` or `!=` operators to improve readability. It's hard to distringuish between `Util` and `!Util`

File: src/test/java/com/axibase/tsd/api/method/metric/MetricCreateOrReplaceTest.java [hunk 4]
```diff
@@ -33,7 +30,7 @@ public void testMetricNameContainsWhiteSpace() throws Exception {
         final Metric metric = new Metric("createreplace metric-1");
 
         Response response = createOrReplaceMetric(metric);
-        assertEquals("Method should fail if metricName contains whitespace", BAD_REQUEST.getStatusCode(), response.getStatus());
+        assertSame("Method should fail if metricName contains whitespace", Response.Status.Family.CLIENT_ERROR, response.getStatusInfo().getFamily());
```
Comment (raipc) [inline src/test/java/com/axibase/tsd/api/method/metric/MetricCreateOrReplaceTest.java]: Replace to `Util.responseFamily(response)`

File: src/test/java/com/axibase/tsd/api/method/AuthenticationTest.java [hunk 5]
```diff
@@ -26,7 +28,7 @@ public void seriesQueryTest() throws Exception {
         List<SeriesQuery> seriesQueryList = Collections.singletonList(new SeriesQuery());
         Response response = SeriesMethod.executeQueryRaw(seriesQueryList, UNKNOWN_USER, UNKNOWN_USER_PASSWORD);
 
-        assertEquals(Response.Status.UNAUTHORIZED.getStatusCode(), response.getStatus());
+        assertSame(Response.Status.Family.CLIENT_ERROR, Util.responseFamily(response));
```
Comment (raipc) [inline src/test/java/com/axibase/tsd/api/method/AuthenticationTest.java]: Please, revert changes in this class. It's important that UNAUTHORIZED status code is returned and not NOT_FOUND or other

File: src/test/java/com/axibase/tsd/api/method/entity/EntityCommandTest.java [hunk 6]
```diff
@@ -167,7 +166,7 @@ public void testIncorrectEnabled(String enabled) throws Exception {
         String command = String.format("entity  e:%s b:%s", entityName, enabled);
         CommandMethod.send(command);
         Response serverResponse = EntityMethod.getEntityResponse(entityName);
-        assertEquals("Bad entity was accepted :: " + command, NOT_FOUND.getStatusCode(), serverResponse.getStatus());
+        assertSame("Bad entity was accepted :: " + command, Response.Status.Family.CLIENT_ERROR, Util.responseFamily(serverResponse));
```
Comment (raipc) [inline src/test/java/com/axibase/tsd/api/method/entity/EntityCommandTest.java]: This method should test for particular status code

Comment (raipc) [reply src/test/java/com/axibase/tsd/api/method/entity/EntityCommandTest.java]: I think, all responses with codes from CLIENT_ERROR family should be tested for particular code. There is a possibility for false-positive assertions in such test methods.

**Comment by rodionos** [comment]
@kirmanak - revert changes that replace specific error codes with families. Only make changes from 200 to 2xx.

**Comment by kirmanak** [comment]
@raipc @rodionos Should I parametrize the tests from `SeriesInsertTest.java` related to wrong URLs? If so, should I create separate PR?

**Pull request merged**